pkg/cloudprovider/providers/cloudstack/metadata.go

```diff
@@ -97,7 +98,11 @@ func (m *metadata) InstanceID(ctx context.Context, name types.NodeName) (string,
 
 // InstanceType returns the type of the specified instance.
 func (m *metadata) InstanceType(ctx context.Context, name types.NodeName) (string, error) {
-	instanceType, err := m.get(metadataTypeInstanceType)
+	instanceTypeCS, err := m.get(metadataTypeInstanceType)
+	// Replace everything that doesn't match the metadata.labels regex
+	re := regexp.MustCompile(`([^A-Za-z0-9][^-A-Za-z0-9_.]*)?[^A-Za-z0-9]`)
```

Comment:
You should not use MustCompile here. It will panic() when regex validation fails.
Since the function can return an error, regexp.Compile() is better.
Instead, do:
```go
re, err := regexp.Compile(`([^A-Za-z0-9][^-A-Za-z0-9_.]*)?[^A-Za-z0-9]`)
if err != nil {
    return "", err
}
```